sdk/identity/Azure.Identity/src/EnvironmentCredentialProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Identity
+{
+    public class EnvironmentCredentialProvider : TokenCredential
```

Comment:
Why some credentials are called "provider" and some "credential". Could we have one naming pattern for all credential types?